main.go

```diff
@@ -173,119 +256,148 @@ func getBackupActionsets(dynamicClient dynamic.Interface, gvr schema.GroupVersio
 					schedule: fmt.Sprintf("%v", backupSchedule),
 				}
 				thisBackup.time, _ = time.Parse(time.RFC3339, fmt.Sprintf("%v", actionMetadata["creationTimestamp"]))
-				backups = append(backups, thisBackup)
+				if thisBackup.schedule == backupConfig.Name {
+					backups = append(backups, thisBackup)
+				}
 			}
 		}
 	}
 	return backups
 }
 
-//determine whether individual backups are required based on max retention dates and their category (daily, weekly, none)
-func categoriseBackups(uncategorisedBackups []backup, dailyBackupsRetained int, weeklyBackupsRetained int) ([]backup, []backup) {
-	var dailyBackups []backup
-	var weeklyBackups []backup
+// determine whether individual backups are required based on max retention dates and their category (daily, weekly, none)
+func categoriseBackups(uncategorisedBackups []backup, backupConfig backupconfig) ([]backup, int) {
+	var categorisedBackups []backup
+	failedBackupCount := 0
+
+	log.Printf("%v: categorising backups\n", backupConfig.Name)
 
-	log.Println("Categorising backups")
+	maxBackupDateTime := time.Now()
 
-	start := time.Now()
+	retentionMinutes, err := strconv.Atoi(backupConfig.Retention.Minutes)
```

Comment:
usually the configuration parsing is done immediately (as if something is wrong the program exits) and - when successful - stored in a struct that can be passed around to whatever func needs it.

Reply:
That is basically what I have - except that by parsing the config from a configmap, the value parsed is a string instead of an int. Is there a way I can convert from string to int whilst unmarshalling the yaml?

Reply:
You can if you write your own unmarshaler: https://codesahara.com/blog/golang-cannot-unmarshal-string-into-go-value-of-type-int/

Reply:
thanks, done this now